# ftp://invisible-island.net/temp/xterm-278-p1.patch.gz
# patch by Thomas E. Dickey <dickey@invisible-island.net>
# created  Tue Nov 27 10:17:11 UTC 2012
# ------------------------------------------------------------------------------
# xstrings.c |    4 ++--
# 1 file changed, 2 insertions(+), 2 deletions(-)
# ------------------------------------------------------------------------------
Index: xstrings.c
--- xterm-278+/xstrings.c	2011-09-11 20:20:12.000000000 +0000
+++ xterm-278-p1/xstrings.c	2012-11-27 10:01:42.000000000 +0000
@@ -214,7 +214,7 @@
     struct passwd *ptr = getpwnam(name);
     Boolean code;
 
-    if (OkPasswd(ptr)) {
+    if (ptr != 0 && OkPasswd(ptr)) {
 	code = True;
 	alloc_pw(result, ptr);
     } else {
@@ -234,7 +234,7 @@
     struct passwd *ptr = getpwuid((uid_t) uid);
     Boolean code;
 
-    if (OkPasswd(ptr)) {
+    if (ptr != 0 && OkPasswd(ptr)) {
 	code = True;
 	alloc_pw(result, ptr);
     } else {
